src/pfe/portal/routes/templates.route.js

```diff
@@ -79,12 +79,24 @@ function sendRepositories(req, res, _next) {
   res.status(200).json(repositoryList);
 }
 
+router.patch('/api/v1/templates/repositories', validateReq, async (req, res) => {
+  const user = req.cw_user;
+  const templateController = user.templates;
+  const requestedOperations = req.body;
+  const operationPromises = requestedOperations.map(operation =>
+    templateController.performOperation(operation)
+  );
+  const operationResults = await Promise.all(operationPromises);
+  res.status(207).json(operationResults);
```

Comment:
sorry, does this mean it will _always_ return 207? Shouldn't that only happen if some fail and some succeed?

in my mind,
all succeed -> 200
some succeed -> 207
all fail -> 400?

if that is annoying to do, we can remove the 207 and just do 200 | 400.

Reply:
(sorry I didn't make this clear on the issue). In my research I've seen a variety of opinions, and I would prefer to use 207 in all situations, to indicate that the response contains a list of sub-statuses, whatever their values. 

It's true that all examples of 207 I've seen describe the sub-codes as being different (e.g. [200, 404, 201]). However, https://httpstatuses.com/207 says that the 207 "response MAY be used in success, partial success and also in failure situations" (e.g. [200, 200, 200]). 

I like using 207 to indicate that the server understood the batch request (hence the main 2xx status), and that there are multiple sub-codes for the user to check.

Reply:
> I like using 207 to indicate that the server understood the batch request (hence the main 2xx status), and that there are multiple sub-codes for the user to check. 

When you put it that way, I think it makes sense.